apps/backend/db_patches/0162_AddStatusActionsLogsTables.sql

```diff
@@ -0,0 +1,27 @@
+DO
+$$
+BEGIN
+  IF register_patch('0162_AddStatusActionsLogsTables.sql', 'Farai Mutambara', 'Add Status Actions Logs', '2024-09-02') THEN
+    BEGIN
+       CREATE TABLE IF NOT EXISTS status_actions_logs(
+            status_actions_log_id serial PRIMARY KEY,
+            connection_id int NOT NULL,
+            action_id int NOT NULL,
+            email_status_action_recipient text DEFAULT NULL,
+            status_actions_by int REFERENCES users (user_id) ON UPDATE CASCADE,
+            status_actions_successful boolean DEFAULT FALSE,
+            status_actions_message text DEFAULT NULL,
+            status_actions_tstamp TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+             FOREIGN KEY (connection_id, action_id) REFERENCES proposal_workflow_connection_has_actions (connection_id, action_id) ON UPDATE CASCADE ON DELETE CASCADE
```

Comment:
Do we really need to delete this log, if the corresponding workflow action is deleted?(ON DELETE CASCADE)

I would recommend that since this is the log, which needs to be persisted for auditing purpose, FK constraint is not really needed to avoid deletion.

Reply:
On our side we may still want to resend even successful emails for example when we edit an email template or user has not received the email for other reasons. We can decide to keep the the record of a status action but  it will have no link to any connection_id, action_id which we require to retry the status action. If it is for auditing we do pass the user information when a status is replayed which will be in the proposal event logs for each proposal affected by the status action.
![image](https://github.com/user-attachments/assets/f5591f84-7079-4c3f-990a-79756f4b3c73)

Reply:
We can also rename this feature if the word logs is misleading.